### ISSUE TITLE
host name gets set to "rancher" always.

### ISSUE THREAD

**Issue by sangeethah** [rancher/os#829]
Rancher server version - v1.0.0-rc3
Rancher os version -  version v0.4.3 / RancherOS (4.2.8-ckt3-rancher)

3 hosts were registered to rancher-server and all the 3 hosts got the host name as  "rancher".

```
[rancher@ip-172-31-23-40 ~]$ hostname
ip-172-31-23-40.us-west-1.compute.internal
[rancher@ip-172-31-23-40 ~]$ 
```

<img width="960" alt="screen shot 2016-03-26 at 7 53 53 am" src="https://cloud.githubusercontent.com/assets/4266958/14060670/d9a4a498-f328-11e5-8029-ef8c855c10a9.png">

**Comment by deniseschannon** [comment]
When I deploy RancherOS in AWS, the hostname is set to `rancher`, which is why it's showing up in Rancher with `rancher`.

Fresh RancherOS setup:

```
[rancher@rancher ~]$ hostname
rancher
```

**Comment by guruvan** [comment]
for a workaround on my hosts I'm simply setting the correct hostname post-cloud-config with /opt/rancher/bin/start.sh

**Comment by guruvan** [comment]
oic now this is not quite working as intended - on AWS, DHCP is resetting the host name to the AWS ip-base hostname (ugh) and /etc/hostname is remaining `rancher` (easily worked around with start.sh) and cloud-config is setting the name correctly on first boot - but is not applied on subsequent boots. ( 0.4.3 )

**Comment by deniseschannon** [comment]
Verified fixed in v0.4.5-rc1
